stock_picking_batch_planner/models/stock_picking_type.py

```diff
@@ -0,0 +1,106 @@
+# Copyright 2025 Moduon Team S.L.
+# License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl-3.0)
+
+from odoo import api, exceptions, fields, models
+
+
+class StockPickingType(models.Model):
+    _inherit = "stock.picking.type"
+
+    plan_batch = fields.Boolean(
+        string="Planned Batches",
+        help="Place Stock Operations in Batches or Waves when planned from the "
+        "Batch of the destiny moves.\n"
+        "Uses same criteria as Auto-Batches to decide how to "
+        "plan batches/waves.",
+    )
+    # Those fields are used to allow Odoo use Auto-Batch logic when planning batches
+    # without changing the behavior of Auto-Batch feature.
+    auto_batch = fields.Boolean(
+        compute="_compute_auto_batch",
+        inverse="_inverse_auto_batch",
+        store=False,
+        readonly=False,
+    )
+    auto_batch_original = fields.Boolean(
+        string="Automatic Batches (original value)",
+        help="Field used to store the original value of `auto_batch` field.",
+    )
+
+    @api.depends_context("plan_batch")
+    @api.depends("auto_batch_original")
+    def _compute_auto_batch(self):
+        """Return True if `plan_batch` is True in context,
+        otherwise return the original value"""
+        plan_batch = bool(self.env.context.get("plan_batch", False))
+        for record in self:
+            record.auto_batch = plan_batch or record.auto_batch_original
+
+    def _inverse_auto_batch(self):
+        """Store the original value of `auto_batch` in `auto_batch_original`.
+        If `auto_batch` is set to True, also set `plan_batch` to True."""
+        for record in self:
+            record.auto_batch_original = record.auto_batch
+            if record.auto_batch_original:
+                record.plan_batch = record.auto_batch_original
+
+    @api.model
+    def _get_view(self, view_id=None, view_type="form", **options):
+        """Change invisible criteria to show items when `plan_batch` or
+        `auto_batch` fields are set."""
+        arch, view = super()._get_view(view_id, view_type, **options)
+        if view_type == "form":
+            for node in arch.xpath("//*"):
+                if (
+                    node.attrib.get("invisible", False)
+                    and "not auto_batch" in node.attrib["invisible"]
+                ):
+                    node.set(
+                        "invisible",
+                        f"not plan_batch and {node.attrib['invisible']}",
```

Comment:
suggestion: this makes sure any order in logic is respected:

```suggestion
                        f"not plan_batch and ({node.attrib['invisible']})",
```

Reply:
Done